### ISSUE TITLE
Make this project buildable again

### ISSUE THREAD

**Issue by thehoneymad** [maplibre/maplibre-java#4]
This project provides some necessary dependencies for MapLibre Java Android SDK as of now. Problems I am seeing here:

1. There is no build guide. Nobody knows how to build it. A quick Gradle build trial failed.
2. README is majorly outdated.
3. Requires Mapbox Access Token in the build process.
4. Still bears a fork relationship to mapbox/mapbox-java. Requires a support ticket to remove the fork relationship.
5. No protection is in place to push to main.
6. Mapbox Service CLI is unnecessary because we are not communicating with Mapbox anymore. I see an opportunity to standardize API communications to any service here. So I will not vote for deleting this now.
7. Same goes for `services-core` package. Possible plugin opportunity. But will require a design.
8. Java Turf port also lives here for some reason. There is absolutely no need for it to live here. So is services-geojson. These packages are published separately. And these are the two packages that gets used in the Android SDK client side.
9. The service APIs are not required for the Android SDK anymore and is bloating the client.
10. No circle CI configuration for a development/test build

_More to come_

**Comment by thehoneymad** [comment]
harping more on number 7, ther service abstraction that we have for Mapbox here covers the following:

1. Directions
2. Refreshed directions. (Refresh directions with movement)
3. Geocoding
4. Geojson
5. Isochrone
6. Map Matching
7. Matrix Routing
8. Route Tiles
9. Speech (Turn by Turn direction)
10. Static Maps
11. Tile Query
12. Java - Turf

**Comment by thehoneymad** [comment]
I will be noting a single comment about the problems I am seeing.

**Comment by thehoneymad** [comment]
Java version is stuck to 1_7 compat. We can update this to Java 11. But I am not sure that will be the right course of action. The Android build might fail. For now, Java 8 will have to do.

**Comment by thehoneymad** [comment]
Current builds are failing due to CheckStyle failures.

**Comment by thehoneymad** [comment]
[ErrorPronPlugin](https://errorprone.info/docs/installation) keeps failing the build. I am not so sure we need this.

**Comment by thehoneymad** [comment]
There is a Nexus publish task that publishes to a [Sonatype Nexus Repository](https://www.sonatype.com/products/nexus-repository). I am not sure where this repository lives and whether we need it at all or not.

**Comment by thehoneymad** [comment]
The gradle script is horribly outdated. Deprecated directives like `classifier` is used.

**Comment by thehoneymad** [comment]
Tons of hanging remote branches. I created https://github.com/maplibre/maplibre-java/issues/5 for it.

**Comment by thehoneymad** [comment]
For now, until advised otherwise, I am following gitflow for git branching, feature maangement, and release management workflow.

**Comment by thehoneymad** [comment]
This is still failing due to checkstyle. I think I can relyably tackle that after #7

**Comment by thehoneymad** [comment]
I am closing this in favour of merging PR https://github.com/maplibre/maplibre-java/pull/8
